rotkehlchen/chain/evm/decoding/cowswap/decoder.py

```diff
@@ -370,6 +370,10 @@ def _decode_normal_claim(self, context: DecoderContext) -> DecodingOutput:
         amount = asset_normalized_value(amount=raw_amount, asset=self.vcow_token)
         airdrop_identifier: Literal['cow_mainnet', 'cow_gnosis'] = 'cow_mainnet' if self.evm_inquirer.chain_id == ChainID.ETHEREUM else 'cow_gnosis'  # noqa: E501
         for event in context.decoded_events:
+            if event.event_type == HistoryEventType.SPEND and event.counterparty != 'gas':
+                event.event_subtype = HistoryEventSubType.PAYMENT
+                event.notes = f'Pay {event.balance.amount} {event.asset.symbol_or_name()} to claim vCOW'  # noqa: E501
+                continue
```

Comment:
![image](https://github.com/user-attachments/assets/68cc462b-49c8-4426-988d-a11eeedc4615)
Checking the transaction there is an attribute claimType that I suspect would tell you if you need to perform this check or not. You can read it by using `hex_or_bytes_to_int(context.tx_log.data[32:64])`. Before entering the loop you can check that value and do something like

```suggestion
            if claim_has_payment and event.event_type == HistoryEventType.SPEND and event.counterparty != 'gas':
                event.event_subtype = HistoryEventSubType.PAYMENT
                event.notes = f'Pay {event.balance.amount} {event.asset.symbol_or_name()} to claim vCOW'  # noqa: E501
                continue
```

Reply:
Forgot it in the review. There in the data of the log event you also have the address. Check too that the address is the same as the location label of the event